src/common/get-node-key.ts

```diff
@@ -0,0 +1,47 @@
+import { NodeContext, Element } from './preboot.interfaces';
+
+/**
+ * Attempt to generate key from node position in the DOM
+ *
+ * NOTE: this function is duplicated in preboot_inline.ts and must be
+ * kept in sync. It is duplicated for right now since we are trying
+ * to keep all inline code separated and distinct (i.e. without imports)
+ */
+export function getNodeKeyForPreboot(nodeContext: NodeContext): string {
```

Comment:
Separated this out to remove duplicate code from old preboot. Now inline preboot and client side preboot share the same function for finding the node.